relayer/chains/cosmos/cosmos_chain_processor.go

```diff
@@ -228,12 +245,28 @@ func (ccp *CosmosChainProcessor) queryCycle(ctx context.Context, persistence *qu
 	ppChanged := false
 
 	for i := persistence.latestQueriedBlock + 1; i <= persistence.latestHeight; i++ {
-		blockRes, err := ccp.cc.Client.BlockResults(ctx, &i)
-		if err != nil {
-			ccp.log.Error("Error getting block results", zap.Error(err))
+		var eg errgroup.Group
+		var blockRes *ctypes.ResultBlockResults
+		var lightBlock *tmtypes.LightBlock
+		eg.Go(func() (err error) {
+			blockRes, err = ccp.cc.Client.BlockResults(ctx, &i)
+			return err
+		})
+		eg.Go(func() (err error) {
+			lightBlock, err = ccp.lightProvider.LightBlock(ctx, i)
+			return err
+		})
+
+		if err := eg.Wait(); err != nil {
+			ccp.log.Error("Error querying block data", zap.Error(err))
 			return nil
 		}
 
+		ccp.latestBlock = provider.LatestBlock{
+			Height: uint64(lightBlock.Height),
+			Time:   lightBlock.Header.Time,
+		}
```

Comment:
At some point, I need a tutorial on how tendermint handles time and timestamps. Is time also part of consensus (like height)? In this case, I'm assuming we can trust this time, but time is indeterminate in distributed systems. It can vary and be forged easily.

Reply:
This timestamp is part of the committed block's header. It is after consensus, so all nodes in the network will return the same timestamp.